app/src/main/java/br/org/cesar/knot_setup_app/view/scan/ScanPresenter.java

```diff
@@ -24,11 +27,58 @@
     private BluetoothWrapper bluetoothWrapper;
     private List<BluetoothDevice> deviceList;
     private UUID service;
+    private ScannerCallback scannerCallback;
+    private ScanCallback scanCallback;
 
     ScanPresenter(ViewModel viewModel, UUID service) {
         this.viewModel = viewModel;
         this.bluetoothWrapper = KnotSetupApplication.getBluetoothWrapper();
         this.service = service;
+
+        scannerCallback = new ScannerCallback() {
+            @Override
+            public void onScanComplete(android.bluetooth.le.ScanResult result) {
+                //Add to adapter list as a connecting option
+                //Check if current device was already inserted on list
+                for (BluetoothDevice device : deviceList) {
+                    if (device.getDevice().getAddress().equals(result.getDevice().getAddress())) {
```

Comment:
if `result.getDevice().getAddress()` has the same value in every iteration, create a variable for it rather than keeping calling method over method everytime